src/test/java/com/mongodb/jdbc/MongoSQLResultSetTest.java

```diff
@@ -285,6 +320,192 @@ void testGetStringValues() throws Exception {
                 mongoSQLResultSet.getUnicodeStream(STRING_COL_LABEL).read(new byte[100], 0, 100));
     }
 
+    @Test
+    public void testGetStringAllTypes() throws Exception {
```

Comment:
I looked into this a bit more and I have a suggestion which I hope can make both of us happy.
I agree that adding an extra test to verify that the String is a valid extended JSON value seems a bit too much.
What we could do is : 
- Extract the values out in separate variables
- Use these snippets in generateRowAllTypes
- Use the them snippet as the expected value

This way, we know that the values are valid extended JSON because they were parsed without any issue by generateRowAllTypes , we verify it's the correct expected value for getString and because testGetObjectToStringMatchesGetString checks that getString == getObject.toString we are good all the way.

Below is a partial example :
`    
public abstract class MongoSQLMock {
...
protected static final String ALL_DOUBLE_COL_VAL = "5";
    protected static final String ALL_TIMESTAMP_COL_VAL = "{\"$timestamp\": {\"t\": 1412180887, \"i\": 1}}";
 ....
    static BsonDocument generateRowAllTypes() {
        String doc =
                    "{"
                        + "    \"all\": {"
                         ....
                         + "        \"timestamp\": "
                        +  ALL_TIMESTAMP_COL_VAL
                        + ","
                        + "        \"long\": "
                        + ALL_DOUBLE_COL_VAL
                        + ","
                       ....
    }
...
}

--------------------------------------------------------------------

class MongoSQLResultSetTest extends MongoSQLMock {
....
    @Test
    public void testGetStringAllTypes() throws Exception {
    ....
            assertEquals( 
                ALL_TIMESTAMP_COL_VAL,
                mongoSQLResultSetAllTypes.getString(ALL_TIMESTAMP_COL_LABEL));
            assertEquals(
               ALL_DOUBLE_COL_VAL, 
               mongoSQLResultSetAllTypes.getString(ALL_LONG_COL_LABEL));
    ....
    }
...
}
`

Reply:
done.